spec/comm.mediawiki

```diff
@@ -146,6 +150,43 @@ The payload for a response has a structure that enables quick error checking.
 }
 </pre>
 
+===Dialogue===
+
+A dialogue type message is initiated the same as a request. The initiator
+will set the ID, which they will then use to identify dialogue messages from the
+other party. The primary difference between a dialogue and a request is that the
+non-initiator must set a `returnid` in their first response. Any subsequent
+dialogue messages from the initiator will use this `returnid` as the message ID.
+
+'''Dialogue initiation'''
+
+<pre>
+{
+	"type": 4,
+	"id": 100,
+	"route" "exchangeinfo",
+	"payload": {"info": "initiator_data_1"},
+}
+</pre>
+
+'''Dialogue initiation response'''
+
+<pre>
+{
+	"type": 4,
+	"id": 100,
+	"returnid": 200,
+	"route" "exchangeinfo",
+	"payload": {"info"; "recipient_data_1"},
+}
+</pre>
+
+The dialogue can then continue, with the initiator addressing messages to the
+specified <code>returnid</code>, and the non-initiator addressing messages to
+the original message ID. The <code>returnid</code> is only expected to be set in
+the first non-initiator message. The initiator will continue to use that ID for
+all subsequent messaging.
```

Comment:
Your implementation description is spot on, and the "all requests" approach is the one I was picturing, but like you said, let the hacking flesh it out. No need to rush this out of draft either. Might be time to just do it, and implement the solution for preimage errors (#597), and then come back to this.

Reply:
May also be a good candidate for the match request, so that the client can get confirmation that their ack was recorded before starting their `'init'` request.  ref: https://github.com/decred/dcrdex/pull/565#discussion_r460186427  Perhaps I'm seeing a logic race where there isn't one, but it seems like an OK is needed back from the server before the maker does its `'init'` request, otherwise the server could be not ready.